### PR TITLE
fix(Core)Rand shuffle seed (RandomBattlegroundSystem,ICC-Marrowgar & Deathwhisper)

### PR DESCRIPTION
**Pull request by GMKyle**
##### CHANGES PROPOSED:
-  Fix spell random target selection of Marrowgar's Bone Spike Graveyard
( http://wotlk.cavernoftime.com/spell=69057) & Lady Deathwhisper's Dominate Mind (http://wotlk.cavernoftime.com/spell=71289)
-  Change Random Battleground Systems' switch shuffle seed 

##### TESTS PERFORMED:
- Build without errors and tested in-game. (Win10 server)

##### HOW TO TEST THE CHANGES:
Note: this behavior it's more evident in Marrowgar encounter. Deathwhisper works better xD
- Enter ICC (any difficulty)
- Engage Marrowgar
- Wait until Bone Spike Graveyard it's casted, notes the random target affected by the spell.
Without this PR Bone Spike Graveyard will be cast on the same target(s) {1 | 3 - 10 | 25}

- Enter ICC (10 H | 25 any difficulty)
- Engage Lady Deathwhisper
- Wait until Dominate Mind it's casted, notes the random target affected by the spell.
Without this PR Dominate Mind will be cast on the same target(s) { 1 |  1|  3 - 10H | 25N | 25 H}

***** IMPORTANT: *****

E.G code
```c++
#include <random>
#include <algorithm>
#include <iterator>
#include <iostream>

int main(int argc, char *argv[])
{
    std::vector<int> v = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10};

    auto rng = std::default_random_engine{rd()};

    std::shuffle(v.begin(), v.end(), std::default_random_engine{});

    copy(v.begin(), v.end(), std::ostream_iterator<int>(std::cout, " "));
    std::cout << "\n";

    /// https://en.cppreference.com/w/cpp/algorithm/random_shuffle

}
```
Always produce the same output. In my case
```c++
8 7 5 6 2 4 10 3 1 9 
```
Include a std::random_device will do the job
```c++
#include <random>
#include <algorithm>
#include <iterator>
#include <iostream>

int main(int argc, char *argv[])
{
    std::vector<int> v = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
    std::random_device rd;
    std::shuffle(v.begin(), v.end(), std::default_random_engine{rd()});

    copy(v.begin(), v.end(), std::ostream_iterator<int>(std::cout, " "));
    std::cout << "\n";

    /// https://en.cppreference.com/w/cpp/algorithm/random_shuffle

}
```
Different outputs. In my case
```c+++
10 5 7 8 6 9 3 1 2 4 
3 9 7 4 6 1 8 5 2 10
5 1 7 3 4 10 9 2 6 8
```

##### KNOWN ISSUES AND TODO LIST:

- [ ]
- [ ] 

##### Target branch(es):
- [x] Master

 

 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by pklloveyou** [comment]
Ready to test?

**Comment by GMKyle** [comment]
@pklloveyou 
Yes

**Comment by pklloveyou** [comment]
Test it tomorrow night.

**Comment by Winfidonarleyan** [comment]
@GMKyle good fix

**Comment by pklloveyou** [comment]
Better than before the PR.

File: src/server/game/Battlegrounds/BattlegroundMgr.cpp [hunk 1]
```diff
@@ -1000,7 +1000,8 @@ void RandomBattlegroundSystem::Update(uint32 diff)
             small.push_back(BATTLEGROUND_AB);
             small.push_back(BATTLEGROUND_SA);
 
-            auto rng = std::default_random_engine{};
+            std::random_device rd;
+            auto rng = std::default_random_engine{rd()};
 
             std::shuffle(big.begin(), big.end(), rng);
```
Comment (LichKing255) [inline src/server/game/Battlegrounds/BattlegroundMgr.cpp]: random_shuffle is not better ?

example:
`std::random_shuffle(big.begin(), big.end());`

Comment (GMKyle) [reply src/server/game/Battlegrounds/BattlegroundMgr.cpp]: before 3a43e8c415cee4041f2a59e9ebc8af721a49be23#diff-85ba4e80476ac43de064d54592507596  ```random_shuffle``` was used, also  ``` shuffle``` is an improvement over ```random_shuffle```

**Pull request merged**